src/emulator/gxm/include/gxm/state.h

```diff
@@ -14,7 +17,81 @@ namespace emu {
     };
 }
 
+template <typename T>
+class Queue
+{
+public:
+
+    uint32_t displayQueueMaxPendingCount_;
+
+    
+    T pop()
+    {
+        std::unique_lock<std::mutex> mlock(mutex_);
+        while(queue_.empty())
+        {
+            condempty_.wait(mlock);
```

Comment:
When I tried to quit the emulator (Cmd-Q) it deadlocked here. It's very inconvenient, but threads also need to wake up and exit when the user wants to quit the emulator.

We need to find a good way around this I think, as it makes multithreading a lot more complex.